### PR TITLE
Add server message handling to dblib

### PR DESCRIPTION
**Pull request by michelka**
Some code taken from example msg_handler defined here:
http://www.freetds.org/userguide/samplecode.htm#SAMPLECODE.ERRORS

### CODE REVIEW

**Comment by michelka** [comment]
The motivation behind this PR is to see more helpful messages when submitting malformed SQL queries.

File: src/dblib_stubs.c [hunk 1]
```diff
@@ -50,6 +50,30 @@ static void raise_fatal(char *msg)
   CAMLreturn0;
 }
 
+static int msg_handler(DBPROCESS *dbproc, DBINT msgno, int msgstate, int severity,
+		       char *msgtext, char *srvname, char *procname, int line)
```
Comment (Chris00) [inline src/dblib_stubs.c]: The handler is given more information than the message, e.g. the `severity` and the `line`.  Any reason not to pass them to the OCaml handler?

**Comment by Chris00** [comment]
@kennknowles I like the idea. Besides my remark on more parameters needing to be made available to OCaml, I think the implementation is fine.

**Comment by Chris00** [comment]
Superseded by https://github.com/kennknowles/ocaml-freetds/pull/8

**Pull request closed without merging**